### PR TITLE
Creation of .travis.yml

### PR DESCRIPTION
**Pull request by Arvin-Esfahani**
The .travis.yml is necessary for Travis CI and having a functional continuous implementation for this GitHub repository.

## Related issue(s)
#98

### CODE REVIEW

**Comment by platisd** [comment]
I suggest you solve the entire issue in one pull request but if you prefer multiple smaller ones it's fine with me. Just let me know which way you prefer and I can approve the request.

**Comment by Arvin-Esfahani** [comment]
I was going to do it in smaller sections but I am sure it'd be better to follow your suggestion. So I will make a pull request when I have the proper .travis.yml file made.

**Comment by platisd** [comment]
The way I would do it is to have a pull request open, like this one, and push to the related branch so I see the status of the Travis job.

**Comment by platisd** [comment]
Let's troubleshoot here. Just write a comment here next time you need help and tag me. :+1: 
There are two errros:
```
  /home/travis/build/DIT112-V19/group-08/Android_Application_for_SmartCar/app/src/main/res/layout/activity_main_page.xml:54: Error: Corresponding method handler 'public void CruiseControl(android.view.View)' not found [OnClick]
          android:onClick="CruiseControl"
```

This one tells you that it cannot find any method to call named `CruiseControl`. With a quick look, I can indeed see that this is the case. I believe there used to be something but most likely not anymore. So it's probably safe to remove that [line](https://github.com/DIT112-V19/group-08/blob/master/Android_Application_for_SmartCar/app/src/main/res/layout/activity_main_page.xml#L54) since it most likely doesn't do anything at the moment. Alternatively, ask the person who implemented this on what was the intention behind it.

**Edit**: Looking at 69315d4caa362af1acb48f3880732ad59263cf94 I can see that at the time there used to be a `CruiseControl` function but not anymore. Specifically, you are not using that particular layout file. So for now, you can just remove that [line](https://github.com/DIT112-V19/group-08/blob/master/Android_Application_for_SmartCar/app/src/main/res/layout/activity_main_page.xml#L54) so to stop the `linter` from complaining.

```
  /home/travis/build/DIT112-V19/group-08/Android_Application_for_SmartCar/app/src/main/AndroidManifest.xml:11: Error: Permission is only granted to system apps [ProtectedPermissions]
      <uses-permission android:name="android.permission.BLUETOOTH_PRIVILEGED" />
```
Here the error message is also pretty clear. You are asking for the `BLUETOOTH_PRIVILEGED` permission but that is only for system apps. So, unless you are running things on a rooted phone or something you are *not* being granted this permission anyway. Therefore, I would bet it's safe to remove [this line](https://github.com/DIT112-V19/group-08/blob/master/Android_Application_for_SmartCar/app/src/main/AndroidManifest.xml#L11) all together.

Alternatively, if you don't want to fix the code, you can always silence the errors by adding the following lines inside the `android` structure in [build.gradle](https://github.com/DIT112-V19/group-08/blob/master/Android_Application_for_SmartCar/app/build.gradle):
```
      lintOptions {
          abortOnError false
      }
```

**Comment by Arvin-Esfahani** [comment]
I am currently facing what seems to be a bug with travis ci or openssl in general on windows. I am trying to sign and create the apk while avoiding publicly holding my api keys. to do that the widely proposed solution was to encrypt the key file with openssl, hold the key and secret of the encryption as environmental variables in Travis CI and then decrypt use and destroy the decrypted key on each run. But for some reason the decryption simply doesn't work. After some googling I found out it's a problem on Windows. Have you encountered this or do you have a workaround? If not I have no choice but to install linux.

**Comment by platisd** [comment]
[Travis CI runs on Ubuntu 16.04 by default](https://blog.travis-ci.com/2019-04-15-xenial-default-build-environment), so I am not sure how Windows is relevant.
I will have a look later, but, do you really need to sign the APK? I mean, it's of course cool to do so, more "professional" and all that, but for the context of this story I personally do not consider it necessary.

**Comment by Arvin-Esfahani** [comment]
The issues seem to be that the encryption done on windows causes a "bad decrypt" error when being decrypted by Travis CI. [This is an example](https://github.com/travis-ci/travis-ci/issues/4746)

**Comment by Arvin-Esfahani** [comment]
The issue seems to be that the encryption done on windows causes a "bad decrypt" error when being decrypted by Travis CI. [This is an example](https://github.com/travis-ci/travis-ci/issues/4746)

**Comment by platisd** [comment]
Very strange and old issue. Well, I am on Linux so if I can help you by running one command or so, let me know.

**Comment by platisd** [comment]
If following [this tutorial](https://android.jlelse.eu/using-travisci-to-securely-build-and-deploy-a-signed-version-of-your-android-app-94afdf5cf5b4) does not work, then skip signing the `.apk` and just deploy the debug version.

File: .travis.yml
Comment (platisd) [inline .travis.yml]: Your `.apk` is most likely located at `$TRAVIS_BUILD_DIR/Android_Application_for_SmartCar/app/build/outputs/apk/debug/app-debug.apk`

Comment (Arvin-Esfahani) [reply .travis.yml]: Thank you for the information, I am still having trouble with keeping the file after the script ends. Apparently there's a cleanup process that deletes it.

File: .travis.yml
Comment (platisd) [inline .travis.yml]: This doesn't really do much. Or to be more exact, it overwrites the apk with a copy of itself!

The file is there, so you should now create a deploy step to deploy it! :+1:

Comment (Arvin-Esfahani) [reply .travis.yml]: I see. I am getting "Skipping a deployment with the releases provider because the current build is a pull request." I am now trying to release to the "The-R-Win-patch-1" branch and see if it makes a difference.

Comment (platisd) [reply .travis.yml]: I think that once you merge it itto master it will be released so remove the `on` part and let's try to merge it.

Comment (Arvin-Esfahani) [reply .travis.yml]: okay I will remove it now.

Comment (Arvin-Esfahani) [reply .travis.yml]: I think it is ready to be merged.

File: .travis.yml
Comment (platisd) [inline .travis.yml]: Can you remove this line? It doesn't do much.

**Review by platisd — CHANGES_REQUESTED** [review]
Almost there!

File: .travis.yml
Comment (Arvin-Esfahani) [reply .travis.yml]: only line 48?

Comment (Arvin-Esfahani) [reply .travis.yml]: I have removed the `- cd $TRAVIS_BUILD_DIR/Android_Application_for_SmartCar/app/build/outputs/apk/debug/`

Comment (platisd) [reply .travis.yml]: Yeah, looks good now. I believe it will create a release once you merge it to master. Let's see... :)

**Review by platisd — APPROVED** [review]

**Comment by platisd** [comment]
Wait for the CI to finish and then click on the "Squash and merge" option, not on the "Merge pull request" to avoid spamming your git history with all those "Update .travis.yml" commits. :grin:

**Comment by platisd** [comment]
Looks like it worked and #98 is fixed! Great effort which I hope will be useful for you in the future! :+1:

**Pull request merged**
